pageserver/src/tenant/gc_block.rs

```diff
@@ -42,6 +43,24 @@ impl GcBlock {
         }
     }
 
+    /// Blocks GC until `duration` has elapsed.
+    ///
+    /// We do this as the leases mapping are not persisted to disk. By delaying GC by default
+    /// length, we guarantee that all the leases we granted before will expire when we run GC for
+    /// the first time after restart / transition from AttachedMulti to AttachedSingle.
+    pub(super) async fn block_for(
+        &self,
+        duration: Duration,
+        cancel: &CancellationToken,
+    ) -> Result<(), Cancelled> {
+        // hold this lock so gc_iteration cannot proceed.
+        let _g = self.blocking.lock().await;
+        match tokio::time::timeout(duration, cancel.cancelled()).await {
+            Ok(_) => Err(Cancelled),
+            Err(_) => Ok(()),
+        }
+    }
```

Comment:
concern: this will also hang up the manual gc blocking enabling.

Reply:
See the new `Storage` based blocking.